tools/check-messages-en.js

```diff
@@ -0,0 +1,133 @@
+const fs = require('fs');
+const path = require('path');
+const { namedTypes: n, visit } = require('ast-types');
+const flowParser = require('flow-parser');
+const { parse } = require('recast');
+
+const messages_en = require('../static/translations/messages_en.json');
+
+// Make a list of files that might contain UI strings, by recursing in src/.
+const possibleUIStringFilePaths = [];
+const kSrcDirName = 'src/';
+function walk(dir, _dirName = '') {
+  let dirent;
+  // eslint-disable-next-line no-cond-assign
+  while ((dirent = dir.readSync())) {
+    // To reduce false negatives, `continue` when nothing in `dirent` can
+    // cause UI strings to appear in the app.
+
+    if (dirent.isFile()) {
+      // Non-JS code, and Flow type definitions in .js.flow files.
+      if (!dirent.name.endsWith('.js')) {
+        continue;
```

Comment:
nit: for a comment like this that's stating the situation we're in when the condition is true, I find it clearer to understand when it's placed inside the conditional rather than outside:

```suggestion
      if (!dirent.name.endsWith('.js')) {
        // Non-JS code, and Flow type definitions in .js.flow files.
        continue;
```

That way the comment describes something that's true at the point where the comment is.